sentry-logback/src/main/java/io/sentry/logback/SentryAppender.java

```diff
@@ -0,0 +1,190 @@
+  private @Nullable Integer flushTimeoutMillis;
+  private @Nullable Integer readTimeoutMillis;
+  private @Nullable Double sampleRate;
+  private @Nullable Boolean bypassSecurity;
+  private @Nullable Boolean debug;
+  private @Nullable Boolean attachThreads;
+  private @Nullable Boolean attachStacktrace;
+
+  private @NotNull final HubAdapter hubAdapter;
+
+  SentryAppender(@NotNull HubAdapter hubAdapter) {
+    this.hubAdapter = hubAdapter;
+  }
+
+  public SentryAppender() {
+    this(HubAdapter.getInstance());
+  }
+
+  @Override
+  public void start() {
+    if (dsn != null) {
+      Sentry.init(
+          options -> {
+            options.setDsn(dsn);
+            Optional.ofNullable(maxBreadcrumbs).ifPresent(options::setMaxBreadcrumbs);
+            Optional.ofNullable(environment).ifPresent(options::setEnvironment);
+            Optional.ofNullable(shutdownTimeoutMillis).ifPresent(options::setShutdownTimeout);
+            Optional.ofNullable(flushTimeoutMillis).ifPresent(options::setFlushTimeoutMillis);
+            Optional.ofNullable(readTimeoutMillis).ifPresent(options::setReadTimeoutMillis);
+            Optional.ofNullable(sampleRate).ifPresent(options::setSampleRate);
+            Optional.ofNullable(bypassSecurity).ifPresent(options::setBypassSecurity);
+            Optional.ofNullable(debug).ifPresent(options::setDebug);
+            Optional.ofNullable(attachThreads).ifPresent(options::setAttachThreads);
+            Optional.ofNullable(attachStacktrace).ifPresent(options::setAttachStacktrace);
+            options.setSentryClientName(BuildConfig.SENTRY_LOGBACK_SDK_NAME);
+            options.setSdkVersion(createSdkVersion(options));
+          });
+    }
+    super.start();
+  }
+
+  @Override
+  protected void append(@NotNull ILoggingEvent eventObject) {
+    hubAdapter.captureEvent(createEvent(eventObject));
+  }
+
+  /**
+   * Creates {@link SentryEvent} from Logback's {@link ILoggingEvent}.
+   *
+   * @param loggingEvent the logback event
+   * @return the sentry event
+   */
+  @SuppressWarnings("JdkObsolete")
+  final @NotNull SentryEvent createEvent(@NotNull ILoggingEvent loggingEvent) {
+    final SentryEvent event = new SentryEvent(new Date(loggingEvent.getTimeStamp()));
+    final Message message = new Message();
+    message.setFormatted(loggingEvent.getFormattedMessage());
+    message.setParams(toParams(loggingEvent.getArgumentArray()));
+    event.setMessage(message);
+    event.setLogger(loggingEvent.getLoggerName());
+    event.setLevel(formatLevel(loggingEvent.getLevel()));
+
+    final ThrowableProxy throwableInformation = (ThrowableProxy) loggingEvent.getThrowableProxy();
+    if (throwableInformation != null) {
+      event.setThrowable(throwableInformation.getThrowable());
+    }
+
+    if (loggingEvent.getThreadName() != null) {
+      event.setExtra("thread_name", loggingEvent.getThreadName());
+    }
+
+    for (Map.Entry<String, String> entry : loggingEvent.getMDCPropertyMap().entrySet()) {
```

Comment:
This is a good candidate to go on `contexts` like:

`event.setContexts("MDC", ....);`